### PR TITLE
2.x: UnicastSubject fix onTerminate

### PR DESCRIPTION
**Pull request by vanniktech**

### CODE REVIEW

File: src/main/java/io/reactivex/subjects/UnicastSubject.java [hunk 1]
```diff
@@ -184,7 +191,7 @@ public void onNext(T t) {
             return;
         }
         if (t == null) {
-            onError(new NullPointerException());
+            onError(new NullPointerException("onNext called with null. Null values are generally not allowed in 2.x operators and sources."));
```
Comment (vanniktech) [inline src/main/java/io/reactivex/subjects/UnicastSubject.java:194]: should not this throw immediately rather than trying to go through onError?

Comment (akarnokd) [reply src/main/java/io/reactivex/subjects/UnicastSubject.java:194]: Yeah, the spec mandates throwing on null input and I think the Processor TCK also expects it. I thought signalling NPE is more graceful with Subjects and FlowableProcessors and shuts down the streams as well whereas a thrown NPE may leave everybody hanging.

**Review by akarnokd — APPROVED** [review]

File: src/main/java/io/reactivex/subjects/UnicastSubject.java (hunk 1 above)
Comment (vanniktech) [reply src/main/java/io/reactivex/subjects/UnicastSubject.java:194]: Alright I'll leave the decision up to you. I'd prefer the fail early approach however delegating it down the stream also works. If you want me to change it I'll do so in a follow up PR.

**Comment by akarnokd** [comment]
Leave this as is and I'll think about it.

**Pull request merged**